src/runtime/methodbinder.cs

```diff
@@ -309,26 +309,38 @@ internal static MethodInfo MatchSignatureAndParameters(MethodInfo[] mi, Type[] g
 
 
                             if (clrtype != null) {
+                                bool typematch = false;
                                 if (pi[n].ParameterType != clrtype) {
                                     IntPtr pytype = Converter.GetPythonTypeByAlias(pi[n].ParameterType);
                                     pyoptype = Runtime.PyObject_Type(op);
                                     Exceptions.Clear();
                                     if (pyoptype != IntPtr.Zero) {
                                         if (pytype != pyoptype) {
-                                            Runtime.Decref(pyoptype);
-                                            margs = null;
-                                            break;
+                                            typematch = false;
                                         }
                                         else {
+                                            typematch = true;
                                             clrtype = pi[n].ParameterType;
                                         }
                                     }
-                                    else {
-                                        Runtime.Decref(pyoptype);
+                                    if (!typematch) {
+                                        // this takes care of enum values
+                                        TypeCode argtypecode = Type.GetTypeCode(pi[n].ParameterType);
+                                        Type typearg = Type.GetType("System." + argtypecode.ToString());
+                                        if (typearg == clrtype) {
+                                            typematch = true;
```

Comment:
@tonyroberts done! thanks for pointing out cleaner solution. I also added this "regression" test.